database/mongoDb.js

```diff
@@ -0,0 +1,214 @@
+    let schema = new mongoose.Schema({
+      house_id: {
+        type: Number,
+        unique: true
+      },
+      type: String,
+      year_built: Number,
+      heating: String,
+      cooling: String,
+      parking: String,
+      lot: Number,
+      days_on_zillow: Number,
+      bedrooms: Number,
+      bathrooms: Number,
+      interiorheating: String,
+      interiorcooling: String,
+      appliances: String,
+      kitchen: String,
+      flooring: String,
+      sqft: Number
+    }, { _id : false });
+    featuresTable = mongoose.model('features', schema);
+    console.log('Connected');
+    featuresTable.countDocuments({}, function(err, count) {
+      if (err) { return handleError(err) } //handle possible errors
+
+      documentCount = count;
+      console.log('The amount of records is: ' + documentCount);
+      cb();
+      //and do some other fancy stuff
+    });
+
+  };
+
+
+
+  const createParams = (numRecords) => {
+    let params = [];
+
+    features.createFeatures(numRecords).forEach(house => {
+      params.push({
+        insertOne : {
+          document: {
+            house_id: documentCount,
+            type:house.type,
+            year_built: house.year_built,
+            heating: house.heating,
+            cooling: house.cooling,
+            parking: house.parking,
+            lot: house.lot,
+            days_on_zillow: house.days_on_zillow,
+            bedrooms: house.bedrooms,
+            bathrooms: house.bathrooms,
+            interiorheating: house.int_heating,
+            interiorcooling: house.int_cooling,
+            appliances: house.appliances,
+            kitchen: house.kitchen,
+            flooring: house.flooring,
+            sqft: house.sqft
+          }
+
+        }
+      });
+
+    });
+
+    return params;
+  }
+
+
+  //function to load feature mock data
+  const loadFeatures = (callback, numRecords = 100, params = []) => {
+    console.log(documentCount);
+    if (params.length === 0) {
+      features.createFeatures(numRecords).forEach(house => {
+        params.push({
+          insertOne : {
+            document: {
+              house_id: ++documentCount,
+              type:house.type,
+              year_built: house.year_built,
+              heating: house.heating,
+              cooling: house.cooling,
+              parking: house.parking,
+              lot: house.lot,
+              days_on_zillow: house.days_on_zillow,
+              bedrooms: house.bedrooms,
+              bathrooms: house.bathrooms,
+              interiorheating: house.int_heating,
+              interiorcooling: house.int_cooling,
+              appliances: house.appliances,
+              kitchen: house.kitchen,
+              flooring: house.flooring,
+              sqft: house.sqft
+            }
+          }
+        });
+      });
+    } else {
+      params.forEach(function (value, index, array) {
+        value.insertOne.document.house_id = ++documentCount;
+      });

```

Comment:
What is the case that params will already have values at this point?

Reply:
Inserting the 31250 records of static Data from 10M Records seed script, otherwise will be created when called from the 100 records seed script which I plan to use for travisCI testing.